examples/sql_bank/main.go

```diff
@@ -53,17 +53,10 @@ func moveMoney(db *sql.DB) {
 
 		switch *transferStyle {
 		case "single-stmt":
-			// TODO(pmattis): Need to add a sub-SELECT here so that we don't create
-			// negative balances:
-			//
-			//   WHERE id IN ($1, $2) AND (SELECT balance >= $3 FROM bank.accounts WHERE id = $1)
-			//
-			// This doesn't currently work as we don't expand placeholders in
-			// sub-SELECTs correctly.
 			update := `
 UPDATE bank.accounts
   SET balance = CASE id WHEN $1 THEN balance-$3 WHEN $2 THEN balance+$3 END
-  WHERE id IN ($1, $2)
+  WHERE id IN ($1, $2) AND true IN (SELECT balance >= $3 FROM bank.accounts WHERE id = $1)
```

Comment:
Maybe, though the example isn't much of an example at this point. Too messy for example purposes in my opinion. The above SQL is (almost) equivalent to the "txn" transfer-style, which is useful for performance comparison purposes. I'd probably also be useful to have a third transfer-style which performs a SELECT and then a single UPDATE statement which updates both rows.